File: Data Structures & Algorithms/validate-parentheses/submission-1.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        pMap = {"}": "{", "]": "[", ")": "("}

        for c in s:
            if c in pMap:
                if len(stack) != 0 and stack[-1] == pMap[c]:
                    stack.pop()
                else:
                    return False
            else:
                stack.append(c)
        
        return True if len(stack) == 0 else False
        

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-1.js
class Solution {
    /**
     * @param {string[]} tokens
     * @return {number}
     */
    evalRPN(tokens) {
        let stack = [];
        for(let char of tokens){
            if(!isNaN(Number(char))){
                stack.push(Number(char))
            }else{
                let a = stack.pop()
                let b = stack.pop()
                let result;
                if(char === '+'){
                    result = b + a
                }else if(char === '-'){
                    result = b - a
                }else if(char === '*'){
                    result = b * a
                }else if(char === '/'){
                    result = Math.trunc(b / a)
                }
                stack.push(result)
            }
        }
        return stack[stack.length - 1]
    }
}
